### ISSUE TITLE
Single Log Out supported by RP?

### ISSUE THREAD

**Issue by bahadorraha** [mitreid-connect/OpenID-Connect-Java-Spring-Server#1310]
Hi,
I reviewed the codes and could not find any code related to RP-initiated SLO request (there was no code submitting request to endSessionEndpoint).
Isn't it implemented yet?

Cheers

**Comment by jricher** [comment]
RP initiated logout is supported in the latest version: https://github.com/mitreid-connect/OpenID-Connect-Java-Spring-Server/blob/master/openid-connect-server/src/main/java/org/mitre/openid/connect/web/EndSessionEndpoint.java

**Comment by bahadorraha** [comment]
Thanks for your reply, but I did not mean support by server!
Your server supports it but can you explain how RP can request SLO using openid-connect-**client** ?

**Comment by jricher** [comment]
Thanks for the clarification. It's not built into the client application library at this time. All you need to do is build the right URL, which you could do using the pieces that are already there, and then send the user there. The first part we could have a utility to help you do that since the library does collect all of that information. The second part is going to be specific to your application.

**Comment by gauravbhanderi** [comment]
Is there any updates on this. I am trying to logout user using the url "/redirect_uri?logout=". This in turn redirects the user to OPenId Connect Server, with another section that says "do you want to logout?". After i hit logout button there, i do get logged out but not redirected to my web app.
In Apache i have the setting for 
OIDCDefaultLoggedOutURL http://<hostname>/logout.html. I even tried the url "/redirect_uri?logout=http://<hostname>/logout.html"

Not sure what i am missing here.

**Comment by jricher** [comment]
Do you have the post-logout redirect URI registered for your client application?

**Comment by gauravbhanderi** [comment]
I can see that setting in the OpenId Connect but with a tag called **NYI**. I did set it to the url from my web app but that ofcourse doesnt take in affect because that feature us **NYI**. BTW i am using version 1.3.2 of MITREid Connect

**Comment by jricher** [comment]
Oh, that tag needs to be updated. If youuse the end session endpoint and pass that URI in, it does get checked. See here: https://github.com/mitreid-connect/OpenID-Connect-Java-Spring-Server/blob/master/openid-connect-server/src/main/java/org/mitre/openid/connect/web/EndSessionEndpoint.java#L181

**Comment by gauravbhanderi** [comment]
http://openidhostname/openid-connect-server-webapp/endsession?id_token_hint=**token_hint**&post_logout_redirect_uri=http%3A%2F%2Fwebapphostname%3A8090

I can see this url on the intermediate(openid connect) logout page, but when i hit logout button there is no redirection to post_logout_redirect_uri

**Comment by gauravbhanderi** [comment]
![image](https://user-images.githubusercontent.com/29868197/35878570-539e7722-0b3e-11e8-9904-753c794d1eb8.png)


Just in case, i am attaching a image.

**Comment by jricher** [comment]
Make sure the post logout URI is an exact string match. The server does no transformations on it, including schemes, ports, and trailing slashes.

**Comment by gauravbhanderi** [comment]
yes, the uri is valid.

**Comment by gauravbhanderi** [comment]
Any other suggestion?

**Comment by gauravbhanderi** [comment]
![nyi](https://user-images.githubusercontent.com/29868197/35881396-47384e2e-0b46-11e8-9a6a-a8e2cf8bf418.png)
I am also attaching an image from the client config where it says that post_logout_redirect_uri is not implemented on the server. Sorry to keep bugging you but I am stuck here with no leads. Any help would be really appreciated.

**Comment by jricher** [comment]
Yes, I know about the "not yet implemented" flag on the UI and already acknowledged it was a bug to have it labeled as such. This feature is implemented in version 1.3.1 and above, as I pointed out previously. At this point your best bet is to put the server into a debugger and watch the call through the end session endpoint to see why it isn't returning. This code has been tested as functional in other environments, so it's something in your setup, which I don't have access to. I've pointed you to the code that is supposed to do this and described what you need to do to have it set up, but now it's on you to figure out why your specific setup isn't doing that. Is it not saving in the session? Are you not sending the string that you think you're sending? (I mean, in your examples I see a `logout.html` tacked on the end of the post-logout URI that you didn't include in your edited example above, but I already told you it's an exact string match so I'm assuming you checked that.)

**Comment by jricher** [comment]
Additionally, none of this discussion is germane to the issue in the original report here which is talking about the RP library, please don't tack on to existing issues.

**Comment by rhinmass** [comment]
How do you get the idtoken-hint? I only see the parse JWT. Not sure how to get the original base64 string which is needed for the token hint?
